### PR TITLE
fix(discover): Type measurements for event search

### PR DESCRIPTION
**Pull request by wmak**
- This updates the event_search grammar to use the builder field type function to determine type since its far more accurate than just checking "is this a measurement"

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -505,15 +505,24 @@ def key_mappings_lookup(self):
...
+            or is_span_op_breakdown(key)
+            or self.builder.get_field_type(key) == "number"
+        )
 
     def is_duration_key(self, key):
         return (
             key in self.config.duration_keys
             or is_duration_measurement(key)
             or is_span_op_breakdown(key)
+            or self.builder.get_field_type(key) == "duration"
```
Comment (edwardgou-sentry) [inline src/sentry/api/event_search.py:520]:
```suggestion
             or self.builder.get_field_type(key) in DURATION_UNITS
```

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -505,15 +505,24 @@ def key_mappings_lookup(self):
...
         )
 
+    def is_size_key(self, key):
+        return self.builder.get_field_type(key) == "size"
```
Comment (edwardgou-sentry) [inline src/sentry/api/event_search.py:524]:
```suggestion
        return self.builder.get_field_type(key) in SIZE_UNITS
```

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (edwardgou-sentry) [reply src/sentry/api/event_search.py:520]: I think get_field_type is returning the unit instead of the actual field type

Comment (edwardgou-sentry) [reply src/sentry/api/event_search.py:520]: Might need both 
```
or self.builder.get_field_type(key) == "duration"
or self.builder.get_field_type(key) in DURATION_UNITS`
```
Not sure if ` self.builder.get_field_type(key) == "duration"` will be used outside of CM

Comment (wmak) [reply src/sentry/api/event_search.py]: good catch!

**Review by edwardgou-sentry — APPROVED** [review]

**Pull request merged**